### PR TITLE
Fix for donut chart when a series only has one non-zero value

### PR DESCRIPTION
**Pull request by djlimbs**
If a data series had only one non-zero value that was not the first element, such as [0, 0, 23, 0, 0], it would incorrectly set the startAngle. This fix checks the series first then updates the startAngle calculation to take that into consideration.

### CODE REVIEW

File: source/scripts/chartist.pie.js [hunk 1]
```diff
@@ -166,6 +166,9 @@
         y: chartRect.y2 + chartRect.height() / 2
       };
 
+      // Check if there is only one non-zero value in the series array.
+      var hasSingleValInSeries = data.series.filter(function(val) { return val !== 0; }).length === 1;
```
Comment (gionkunz) [inline source/scripts/chartist.pie.js]: Cool stuff :-) Thanks for working on this and using Array extras! In Chartist we follow the convention to write function bodies on a new line. Can you put the return statement on a new line something like this?

``` javascript
var hasSingleValInSeries = data.series.filter(function(val) {
  return val !== 0; 
}).length === 1;
```

Comment (djlimbs) [inline source/scripts/chartist.pie.js]: Yes, no problem, I'll update it and re-submit. Thanks!

**Comment by gionkunz** [comment]
Just tested this and it looks good :+1:

**Comment by djlimbs** [comment]
Awesome!

**Pull request merged**